app/src/main/java/ar/com/wolox/android/example/ui/auth/login/AuthPresenter.kt

```diff
@@ -31,6 +42,24 @@ class AuthPresenter @Inject constructor(
         loginRequest(LoginData(email, password))
     }
 
+    @RequiresApi(Build.VERSION_CODES.M)
+    fun isOnline(): Boolean {
+        val result: Boolean
+        val connectivityManager =
+            context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
```

Comment:
I would instantiate the Connectivity Manager in the fragment, and then pass it to the presenter. 
@ffasitella @rodrigo103

Reply:
I totally agree. As Lisandro said, it's not a good idea to have access to the context inside the presenter.